File: src/server/actions/dictionary.ts
'use server';

import { type Definition } from '../../types/definition';

export async function getDefinition(word: string) {
  const response = await fetch(
    `https://api.dictionaryapi.dev/api/v2/entries/en/${word}`,
    {
      method: 'GET',
      next: { tags: ['update-definition'] },
    },
  );
  try {
    const data = await response.json();
    const definition: Definition = data[0];
    return definition;
  } catch (error) {
    console.error(error);
  }
}

File: services/ink/src/api/dictionary.rs
use crate::errors::DictionaryError;
use crate::models::definition::Definition;
use axum::{Json, Router, routing::get, extract::Path};

pub fn router() -> Router {
  Router::new().route("/dictionary/:word", get(get_definition))
}

async fn get_definition(
  Path(word): Path<String>,
) -> Result<Json<Vec<Definition>>, DictionaryError> {
  let url = format!(
    "https://api.dictionaryapi.dev/api/v2/entries/{}/{}",
    "en", word
  );
  let response = reqwest::get(url).await?.json::<Vec<Definition>>().await?;

  Ok(Json(response))
}
